kratos/mappers/mapper.h

```diff
@@ -110,6 +111,10 @@ class Mapper
         const Variable< array_1d<double, 3> >& rDestinationVariable,
         Kratos::Flags MappingOptions) = 0;
 
+    /// @brief Mapping interface for @ref Expression s.
+    virtual Expression::Pointer Map(Expression::Pointer pOriginExpression,
+                                    Kratos::Flags MappingOptions) = 0;
```

Comment:
this differs from the other interfaces, which take both input and output as argument, and return nothing
Any objections to making it consistent?

Reply:
I would, but unfortunately [`Expression`](https://github.com/KratosMultiphysics/Kratos/blob/5d3b297575ec0accb88648c14d718a9844c50a32/kratos/expression/expression.h#L38)s don't allow that. @sunethwarna decided to implement them in a way that they can't be mutated after construction.

Reply:
:O In this case, if we are to be consistent with the other interface methods, we could have the following (not a big fan, but it is a possibility):
```c++
virtual void Map(LiteralFlatExpression<double>& rOutputExpression,  const Expression& rOriginExpression, Kratos::Flags MappingOptions) = 0;
```

Any reason why `Expression::Pointer` is passed not the `const Expression&` ?

Reply:
I don't like that either. It will prevent reverse mapping.

Reply:
One possibility is to pass a mutable reference to the destination expression, and reset it to point to a different expression. It's not a great solution either because other pointers pointing to the original expression won't get updated.